src/devices/cpu/z80/z80.cpp

```diff
@@ -181,20 +173,27 @@ void z80_device::add_a(u8 value)
  ***************************************************************/
 void z80_device::adc_a(u8 value)
 {
-	u32 ah = AF & 0xff00, c = AF & 1;
-	u32 res = (u8)((ah >> 8) + value + c);
-	set_f(SZHVC_add[(c << 16) | ah | res]);
+	const bool c = F & CF;
```

Comment:
Change to int?
math with a boolean is kind of icky.

CF is 1 on Z80, but maybe you could add assert(CF == 0x01); to z80.cpp to emphasize it?

Reply:
I did'n get assert purpose. CF is carry flag positioned at 0th bit of F

Reply:
Since CF == 1, it doesn't need ?1:0;
My point for adding an assert was, in some parts of the code you rely on the flag positions with math, an assert emphasizes that.